ground_truth_labeling_jobs/ground_truth_conversion_scripts/GT_semantic_segmentation_to_COCO.ipynb

```diff
@@ -0,0 +1,405 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Semantic Segmentation Job output conversion to COCO format\n",
+    "This notebook walks through converting the output from SageMaker Ground Truth Semantic segmentation task into Common Objects in Context (COCO) format. The output manifest of the semantic segmentation task contains a reference to a PNG file masks for the objects that has been annotated and saved in an Amazon S3 bucket. In this notebook, I will download a sample manifest file, convert the mask into a pixelated format as in the COCO format."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Install Pre-requisites"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install opencv-python pycocotools"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get the output.manifest file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "import os\n",
+    "\n",
+    "JOBNAME = '<LABELING JOB NAME>' #Replace it with the labeling job name\n",
+    "REGION = '<REGION>' #Replace it with the job region\n",
+    "client = boto3.client('sagemaker',region_name=REGION)\n",
+    "response = client.describe_labeling_job(LabelingJobName= JOBNAME)\n",
+    "file = response['LabelingJobOutput']['OutputDatasetS3Uri']\n",
+    "output_manifest = os.path.basename(file)\n",
+    "!aws s3 cp $file ./"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Read the manifest file line by line and create \"Images\" key"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "data_objs  = []\n",
+    "input_files = []\n",
+    "images = []\n",
+    "file_name = os.path.join(\"./\", output_manifest)\n",
+    "\n",
+    "#Assuming all images are reshaped to have the same dimensions\n",
+    "height = 3956\n",
+    "width = 5280\n",
+    "\n",
+    "#Get data objects from the output manifest\n",
+    "with open(file_name) as out_manifest:\n",
+    "    for line in out_manifest:\n",
+    "        data_objs.append(json.loads(line))\n",
+    "\n",
+    "#Get the input images filenames\n",
+    "for line in data_objs:\n",
+    "        input_files.append(os.path.basename(line['source-ref']))\n",
+    "\n",
+    "dir_path = os.path.dirname(data_objs[0]['source-ref'])\n",
```

Comment:
This finds the s3 folder of the first object in the manifest. Later on, you use it to download a test file. However, there is no guarantee that all the objects in the manifest are in the same s3 folder. As a result, the download fails for me. For example, if my first object is at 
`'s3://some-bucket/foo/example1.png'`
and the second is at
`'s3://some-bucket/bar/example2.png'`,
the notebook might try to download `'s3://some-bucket/foo/example2.png'`, which doesn't exist.

Reply:
This is resolved by adding the path to the json file and preserving the path in the local dir as well